### ISSUE TITLE
for premium_lrs make sure we disable bariers

### ISSUE THREAD

**Issue by Mpdreamz** [elastic/azure-marketplace#40]
as specified here: https://azure.microsoft.com/en-us/documentation/articles/storage-premium-storage/

**Comment by russcam** [comment]
As the linked Azure article indicates, 

> For Premium Storage disks with cache setting “ReadWrite”, barriers should be enabled for durability of writes.

barriers should be disabled only when cache setting is `“ReadOnly”` or `“None”` (the default for data disks is `"ReadOnly"` but for OS disks is `"ReadWrite"`).

From https://azure.microsoft.com/en-us/documentation/articles/storage-premium-storage-performance/

> **ReadWrite**
By default, the OS disks have ReadWrite caching enabled. We have recently added support for ReadWrite caching on data disks as well. If you are using ReadWrite caching, you must have a proper way to write the data from cache to persistent disks. For example, SQL Server handles writing cached data to the persistent storage disks on its own. Using ReadWrite cache with an application that does not handle persisting the required data can lead to data loss, if the VM crashes.

In summary, we should disable barriers for PREMIUM_LRS data disks, but it would also be worth us performing some investigation on the different cache settings options for Premium Locally Redundant Storage, similar to http://codehardblog.azurewebsites.net/couchbase-on-azure-does-readwrite-caching-help-2/

**Comment by niemyjski** [comment]
What's the status of this.
